### DISCUSSION TITLE
Dictionary TryGetValue(TKey, TValue) - Unhandled exception.

### DISCUSSION THREAD

**Discussion by Was85** [dotnet/runtime#81357] [General] [answered]
I found it weird to get an Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'key') when I pass null into a function called **Try**GetValue(TKey, TValue).

 I know it's stated clearly in the Docs that this will happen when pass null. However, I believe this should be changed and returned as false instead of throwing an exception.

**Comment by koszeggy** [comment, marked as answer]
>  I believe this should be changed

Modifying this behavior would be a breaking change. And considering that keys cannot be `null` this behavior makes sense.

But you can always define a new extension method, eg.

```cs
public static bool MyTryGetValue<TKey, TValue>(this IDictionary<TKey, TValue> dict,
    TKey key, [MaybeNullWhen(false)]out TValue value)
{
    if (key == null)
    {
        value = default;
        return false;
    }

    return dict.TryGetValue(key, out value);
}
```

And just a side note: I needed to create an [`AllowNullDictionary`](https://github.com/koszeggy/KGySoft.CoreLibraries/blob/master/KGySoft.CoreLibraries/Collections/AllowNullDictionary.cs) type once. But please note that its [`TryGetValue`](https://github.com/koszeggy/KGySoft.CoreLibraries/blob/f5af53d716edd45d35e520c5e092ede71353afbb/KGySoft.CoreLibraries/Collections/AllowNullDictionary.cs#L183) explicitly contains a `[CanBeNull]` attribute for the `key` parameter, which triggered CS8767 (Nullability of reference types in type of parameter doesn't match implicitly implemented member) that I needed to [suppress](https://github.com/koszeggy/KGySoft.CoreLibraries/blob/f5af53d716edd45d35e520c5e092ede71353afbb/KGySoft.CoreLibraries/Collections/AllowNullDictionary.cs#L34), which indicates that the current behavior is very much expected.

**Comment by HaloFour** [comment]
Also, this would be a request to the [runtime repository](https://github.com/dotnet/runtime), which handles the base class libraries and APIs for .NET.
